storage/applicationinsights/src/main/java/zipkin/storage/applicationinsights/ApplicationInsightsSpanConsumer.java

```diff
@@ -68,6 +85,24 @@ public void accept(List<Span> spans) {
         }
         String res = gson.toJson(jsonElement);
         String msg = "{ \"Span\":" + res + "}";
+        //data model changes
+        telemetry.getContext().getOperation().setId(traceId);
+        telemetry.getContext().getOperation().setName(span.name);
+
+        for (Annotation annotation : span.annotations) {
+
+          if (annotation.value.equalsIgnoreCase(Constants.CLIENT_SEND)) {
+            String spanName = span.name !=null && !span.name.isEmpty()?span.name:Constants.CLIENT_SEND;
```

Comment:
Yes, they suggest logging a  [binary annotation](https://github.com/openzipkin/zipkin/blob/master/zipkin/src/main/java/zipkin/Constants.java#L150) and have an open [issue] (https://github.com/openzipkin/openzipkin.github.io/issues/52). We shall discuss further how we set AI flag.